src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/OfflineAuthenticationTest.java

```diff
@@ -174,54 +178,123 @@ void GIVEN_clientDevice_WHEN_verifyingItsIdentity_THEN_pemStored() throws Interr
-       X509Certificate clientBCrt = createClientCertificate(
-               rootCA, "AWS IoT Certificate", clientBKeyPair.getPublic(), rootKeyPair.getPrivate());
-
-       String clientAPem = CertificateHelper.toPem(clientACrt);
-       iotAuthClientFake.activateCert(clientAPem);
-       String clientBPem = CertificateHelper.toPem(clientBCrt);
-       iotAuthClientFake.activateCert(clientBPem);
-       when(networkStateMock.getConnectionStateFromMqtt()).thenReturn(NetworkState.ConnectionState.NETWORK_UP);
-
-       // When
-       givenNucleusRunningWithConfig("config.yaml");
-
        Instant now = Instant.now();
        Runnable resetClock = mockInstant(now.toEpochMilli());
 
-       // Simulate some client components (like Moquette) verifying some certificates
-       ClientDevicesAuthServiceApi api = kernel.getContext().get(ClientDevicesAuthServiceApi.class);
-       api.verifyClientDeviceIdentity(clientAPem);
-       api.verifyClientDeviceIdentity(clientBPem);
+       // Given
+       givenCoreDeviceIsOnline();
+       List<String> clientDevicePems = givenClientDevicesWithCertificates(2);
 
-       CertificateRegistry certRegistry = kernel.getContext().get(CertificateRegistry.class);
-       Certificate ogCertA = certRegistry.getCertificateFromPem(clientAPem).get();
-       Certificate ogCertB = certRegistry.getCertificateFromPem(clientBPem).get();
-       assertEquals(ogCertA.getStatusLastUpdated().toEpochMilli(), now.toEpochMilli());
-       assertEquals(ogCertB.getStatusLastUpdated().toEpochMilli(), now.toEpochMilli());
+       // When
+       givenNucleusRunningWithConfig("config.yaml");
+       givenClientDevicesVerifyTheirIdentity(clientDevicePems, now);
 
        Instant anHourLater = now.plusSeconds(60 * 60);
        resetClock.run();
-       mockInstant(anHourLater.toEpochMilli());
+       resetClock = mockInstant(anHourLater.toEpochMilli());
 
        BackgroundCertificateRefresh backgroundRefresh = kernel.getContext().get(BackgroundCertificateRefresh.class);
        assertTrue(backgroundRefresh.isRunning(), "background refresh is not running");
        backgroundRefresh.run(); // Force a run because otherwise it is controlled by a ScheduledExecutorService
        kernel.getConfig().waitConfigUpdateComplete();
 
        // Then
-       Certificate certA = certRegistry.getCertificateFromPem(clientAPem).get();
-       Certificate certB = certRegistry.getCertificateFromPem(clientBPem).get();
+       CertificateRegistry certRegistry = kernel.getContext().get(CertificateRegistry.class);
+       Certificate certA = certRegistry.getCertificateFromPem(clientDevicePems.get(0)).get();
+       Certificate certB = certRegistry.getCertificateFromPem(clientDevicePems.get(1)).get();
 
        assertEquals(certA.getStatusLastUpdated().toEpochMilli(), anHourLater.toEpochMilli());
        assertEquals(certB.getStatusLastUpdated().toEpochMilli(), anHourLater.toEpochMilli());
+       resetClock.run();
```

Comment:
if this is important to run at the end of tests, put it in an afterEach. Failed assertions will mean that this code would never execute.